### DISCUSSION TITLE
Negative strut?

### DISCUSSION THREAD

**Discussion by sqwishy** [YaLTeR/niri#208] [Ideas]
In my config I have these huge gaps -- like 32px -- it's embarrassing to be honest. Waybar is narrow enough to fit inside that region at the top of the display. So at the top of the display I want a 32px gap from the edge of the display, not from waybar. From waybar the gap is like 6px or something.

It looks like the gap is just a single value, as opposed to reducing the top gap independently (like how struts can be set for left right top bottom).

But it appears that using a negative strut to remove the gap from the top also does what I want. Changing that type from `u16` to `i32` works and removes `into()` and `as i32` in `compute_working_area()`.

Anyway, just some feedback about my use case and how I chose to support it.

**Comment by YaLTeR** [comment]
Negative struts would be kinda weird ngl, but I see your use-case. For waybar, you can probably configure it to not use layer-shell exclusive area, then it won't contribute to the struts. `"mode": "overlay"` could be it?

**Reply by sqwishy** [reply]
Yeah actually that works fine instead. Good idea, thanks!
